src/state/apis/zapper/client.ts

```diff
@@ -549,14 +549,137 @@ export enum ZapperGroupId {
   Farm = 'farm',
 }
 
+const MEDIA_FILETYPE = ['mp4', 'png', 'jpeg', 'jpg', 'gif', 'svg', 'webp'] as const
+export type MediaFileType = typeof MEDIA_FILETYPE[number]
+export type MediaType = 'video' | 'image'
+
+export const getMediaFileType = (mediaUrl: string | undefined): MediaFileType | undefined => {
+  if (!mediaUrl) return undefined
+  const mediaFiletype = mediaUrl.split('.').pop()
+  return mediaFiletype as MediaFileType | undefined
+}
+
+export const getMediaType = (mediaUrl: string | undefined) => {
+  const mediaFileType = getMediaFileType(mediaUrl)
+  if (!mediaFileType) return undefined
+
+  if (mediaFileType === 'mp4') return 'video'
+  return 'image'
+}
+
+const mediaSchema = z.object({
+  type: z.string(),
+  originalUrl: z
+    .string()
+    .url()
+    .refine(url => {
+      const mediaFileType = getMediaFileType(url)
+      if (!mediaFileType) return false
+      return MEDIA_FILETYPE.includes(mediaFileType as MediaFileType)
+    }, 'Media filetype not supported'),
+})
+
+const socialLinkSchema = z.object({
+  name: z.string(),
+  label: z.string(),
+  url: z.string().url(),
+  logoUrl: z.string().url(),
+})
+
+const statsSchema = z.object({
+  hourlyVolumeEth: z.number(),
+  hourlyVolumeEthPercentChange: z.nullable(z.number()),
+  dailyVolumeEth: z.number(),
+  dailyVolumeEthPercentChange: z.nullable(z.number()),
+  weeklyVolumeEth: z.number(),
+  weeklyVolumeEthPercentChange: z.nullable(z.number()),
+  monthlyVolumeEth: z.number(),
+  monthlyVolumeEthPercentChange: z.nullable(z.number()),
+  totalVolumeEth: z.number(),
+})
+
+const fullCollectionSchema = z.object({
+  name: z.string(),
+  network: z.string(),
+  description: z.string(),
+  logoImageUrl: z.string().nullable(),
+  cardImageUrl: z.string().nullable(),
+  bannerImageUrl: z.string().nullable(),
+  nftStandard: z.string(),
+  floorPriceEth: z.string().nullable(),
+  marketCap: z.string().optional(),
+  openseaId: z.string().nullable(),
+  socialLinks: z.array(socialLinkSchema),
+  stats: statsSchema,
+})
+
+const nftCollectionSchema = z.object({
+  balance: z.string(),
+  balanceUSD: z.string(),
+  collection: fullCollectionSchema,
+})
+
+const cursorSchema = z.string()
+
+const v2NftBalancesCollectionsSchema = z.object({
+  items: z.array(nftCollectionSchema),
+  cursor: cursorSchema,
```

Comment:
I think `cursor` can be `undefined` if the `items` array is empty.

I'm getting the below when switching between Metamask wallets (even though both wallets have NFTs that are shown):

<img width="813" alt="Screenshot 2023-04-06 at 1 19 03 pm" src="https://user-images.githubusercontent.com/97164662/230263889-d53d95ed-bf2d-4189-898d-a0ea9cb1bb13.png">

As far as I can tell, it seems what is happening is that the accountId (the `addresses` param) changes first, then the RTK query fires (returning `{ items: [] }`, then the `collectionAddresses` updates, causing the RTK query to fire again, this time with the correct params, resulting in a valid response.

Reply:
Absolutely can - d19324bc8. Confirmed with an empty account, and same goes for `/v2/nft/user/tokens` 
<img width="923" alt="image" src="https://user-images.githubusercontent.com/17035424/230324185-aa83a333-b937-40c4-9811-7ae98defa653.png">

<img width="912" alt="image" src="https://user-images.githubusercontent.com/17035424/230323741-b9f544e0-1b2e-4e4b-94d7-50afc7bf89ef.png">